src/test/java/com/pngencoder/PngEncoderTest.java

```diff
@@ -211,7 +212,24 @@ public void testPredictorEncodingWithImage() throws IOException {
     }
 
     @Test
-    public void testpredictorEncodingCompareSize() throws IOException {
+    public void testIndexedEncodingWithImage() throws IOException {
+        final BufferedImage bufferedImage = ImageIO
+                .read(Objects.requireNonNull(PngEncoderTest.class.getResourceAsStream("/png-encoder-logo.png")));
+
+        PngEncoder pngEncoder = new PngEncoder()
+                .withBufferedImage(bufferedImage)
+                .withCompressionLevel(1)
+                .withTryIndexedEncoding(true);
+
+        /*
+         * We can not compare the raw image rgba pixels, as the indexed encoder "normalizes" pixels with an alpha value of
+         * zero to 0
+         */
+        validateImage(PngEncoderBufferedImageType.TYPE_CUSTOM, bufferedImage, pngEncoder);
```

Comment:
See https://github.com/pngencoder/pngencoder/blob/dc9967ecaf015812c665f0f4ae4c154b850847ce/src/test/java/com/pngencoder/PngEncoderTestUtil.java#L90. You can change the util if you want!